src/app/qgshandlebadlayers.cpp

```diff
@@ -526,6 +527,7 @@ void QgsHandleBadLayers::autoFind()
 {
   QDir::setCurrent( QgsProject::instance()->absolutePath() );
   QgsProject::instance()->layerTreeRegistryBridge()->setEnabled( true );
+¨
```

Comment:
I assume you did not compile this before opening this PR. Can I ask you to please do that?

Reply:
I'll try to get it done today, I borked my dev environment yesterday and I get the 'invalid CMAKE GENERATOR' after cloning my kit, and not every time I run cmake I get that error. I have no clue wtf is going on with qt creator and cmake but any changes just break everything and I'm forced to reinstall stuff until everything works again.

Reply:
Your work is really appreciated.
However, opening a PR for review without prior local testing may seem like asking for other's time for unfinished things.

* If you are mostly looking for conceptual inputs, please clearly state those questions and we can discuss those and no one needs to spend time on unfinished code.
* If you have a working implementation, open a pull request that compiled for you and if possible add a screencast to the PR so we can easily discuss on UX and implementation ends.
* If you have a broken development environment, try to fix it or send a mail to the dev list and work on fixing it before submitting code for review.

I hope this helps for the future.

Reply:
At this stage, as its building (finished fixing my environment just now). I mostly wanted to know the following elements:

- Is that dialog fine to use within QGIS or are there better option?

- Are there templates or other things that must be applied to those dialogs in order to maintain the QGIS look?

Otherwise I should be able to provide a gif as soon as it finishes building.

I didn't think about building at first as I thought those were only minor changes to the code base and were pretty standard and I could compile afterward as compiling takes some time in my VM.

Reply:
> At this stage, as its building (finished fixing my environment just now). I mostly wanted to know the following elements:
> 
>     * Is that dialog fine to use within QGIS or are there better option?
> 

I think it's fine, generally speaking always check first if is there any specific `Qgs` subclass, but in this case I think there isn't any,

>     * Are there templates or other things that must be applied to those dialogs in order to maintain the QGIS look?

No

> 
> 
> Otherwise I should be able to provide a gif as soon as it finishes building.
> 
> I didn't think about building at first as I thought those were only minor changes to the code base and were pretty standard and I could compile afterward as compiling takes some time in my VM.

This is generally speaking very bad practice, it's been already told you, please make sure your code builds correctly, pass it through prepare-commit hooks and try to wait until you've done all final adjustments before submitting a PR.

If you don't follow community member advice people will just stop reviewing your PRs.